### PR TITLE
[UPD] feat(workflow): New wokflow definition

### PR DESCRIPTION
**Pull request by Pourbaix**
Changed the workflow definition to have only the pdfattestation + review steps. Also, when a reviewer rejects a submission, it is set to the 'withdrawn' state.

### CODE REVIEW

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+/**
+ * Custom review action for master theses.
+ * (mainly a copy of @org.dspace.xmlworkflow.state.actions.processingaction.ReviewAction).
+ * This Action contains only 2 outputs: Accepted or Rejected
+ * In the case 'Accepted', we continue the workflow 
+ * In the case 'Rejected', we change the state of the workflow item to 'Withdrawn'
+ */
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: COSMETIC : 
Don't hesitate to set `@author` and `@version` annotation into class docstring.
```
 * @author Michaël Pourbaix <michael.pourbaix@uclouvain.be>
 * @version $Revision$
 ```

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+public class ThesisReviewAction extends ReviewAction {
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: COSMETIC : Classname could be "UCLouvainThesisReviewAction" ?

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    protected static final String SUBMIT_REJECT = "submit_reject";
+    protected static final String RETURN_TO_POOL = "return_to_pool";
+    protected static final String REJECT_REASON = "reason";
+
+    @Autowired(required = true)
+    protected ClaimedTaskService claimedTaskService;
+    @Autowired(required = true)
+    protected ItemService itemService;
+    @Autowired
+    protected XmlWorkflowService xmlWorkflowService;
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: remove too (and test)

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public static final int REJECT_PAGE = 1;
+
+    private static final String SUBMITTER_IS_DELETED_PAGE = "submitter_deleted";
+
+    public static final String SUBMIT_EDIT_METADATA = "submit_edit_metadata";
+    public static final String SUBMIT_CANCEL = "submit_cancel";
+    protected static final String SUBMIT_APPROVE = "submit_approve";
+    protected static final String SUBMIT_REJECT = "submit_reject";
+    protected static final String RETURN_TO_POOL = "return_to_pool";
+    protected static final String REJECT_REASON = "reason";
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: remove too: already defined into super classes

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ * (mainly a copy of @org.dspace.xmlworkflow.state.actions.processingaction.ReviewAction).
+ * This Action contains only 2 outputs: Accepted or Rejected
+ * In the case 'Accepted', we continue the workflow 
+ * In the case 'Rejected', we change the state of the workflow item to 'Withdrawn'
+ */
+public class ThesisReviewAction extends ReviewAction {
+    public static final int MAIN_PAGE = 0;
+    public static final int REJECT_PAGE = 1;
+
+    private static final String SUBMITTER_IS_DELETED_PAGE = "submitter_deleted";
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: Remove these 3 lines:
  * (logicaly) Constants are defined into super class and should be used from it. The must not be redefine in this class.
  * (pratically) None of these constants are used in this class --> useless (a good IDE will prevent you about this problem)
  
--> Also check all import and remove unused imported classes.

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Process result when option {@link this#SUBMIT_REJECT} is selected.
+     * - Sets the reason and workflow step responsible on item in dc.description.provenance
+     * - Send workflow back to the submission
+     * If reason is not given => error
+     */
+    @Override
+    public ActionResult processRejectPage(Context c, XmlWorkflowItem wfi, HttpServletRequest request) throws SQLException, AuthorizeException, IOException {
+        String reason = request.getParameter(REJECT_REASON);
+        if (reason == null || 0 == reason.trim().length()) {
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: `0 == reason.trim().length()` --> `reason.isBlank()`

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+    private Item archive(Context context, XmlWorkflowItem wfi) throws SQLException, IOException, AuthorizeException {
+        Item item = wfi.getItem();
+
+        workflowItemRoleService.deleteForWorkflowItem(context, wfi);
+
+        installItemService.installItem(context, wfi);
+
+        itemService.clearMetadata(context, item, WorkflowRequirementsService.WORKFLOW_SCHEMA, Item.ANY, Item.ANY, Item.ANY);
+        itemService.update(context, item);
```
Comment (zannkukai) [inline dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: COSMETIC : remove additionnal blank lines

**Review by zannkukai — CHANGES_REQUESTED** [review]

File: dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java (hunk 5 above)
Comment (Pourbaix) [reply dspace-uclouvain/src/main/java/org/dspace/uclouvain/xmlworkflow/actions/ThesisReviewAction.java]: My IDE is shit, it didn't show me ... :face_with_spiral_eyes: 
VSCode => :wastebasket:

**Review by zannkukai — APPROVED** [review]

**Pull request merged**